src/Adapter/Carrier/Repository/CarrierRangeRepository.php

```diff
@@ -0,0 +1,249 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Carrier\Repository;
+
+use Carrier;
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Query\QueryBuilder;
+use IteratorAggregate;
+use PrestaShop\PrestaShop\Core\Domain\Carrier\Exception\CarrierConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Carrier\Exception\CarrierException;
+use PrestaShop\PrestaShop\Core\Domain\Carrier\ValueObject\CarrierId;
+use PrestaShop\PrestaShop\Core\Domain\Carrier\ValueObject\CarrierRange;
+use PrestaShop\PrestaShop\Core\Domain\Carrier\ValueObject\CarrierRangePrice;
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint;
+
+/**
+ * Provides access to carrier range data source
+ */
+class CarrierRangeRepository
+{
+    public function __construct(
+        protected readonly Connection $connection,
+        protected readonly string $dbPrefix,
+        protected readonly CarrierRepository $carrierRepository,
+    ) {
+    }
+
+    /**
+     * @param CarrierId $carrierId
+     *
+     * @return CarrierRange[]
```

Comment:
I wonder if we should return a "collection" object instead of an array of ranges Mostly because I think the ranges alone lack one information: the shipping method We need to know if the returned ranges are based on price or weight And ideally we shouldn't need to fetch the carrier on the side to get this info

In my idea the collection look like this
```
interface CarrierRangesCollectionInterface {
    public function getShippingMethod(): string; // Value from the VO ShippingMethod
    public function getRanges(): array; // CarrierRange[]
}
```

Reply:
Ok, maybe change the naming of "shippingMethod" in DTO too and create Collection with static method to build the collection.